doc/reference/configuration/configuration_reference.rst

```diff
@@ -2889,6 +3004,40 @@ This section describes configuration parameters related to roles.
 
     **Since:** :doc:`3.0.0 </release/3.0.0>`.
 
+    Specify a role's configuration.
+    This option accepts a role name as the key and a role's configuration as the value.
+    To specify the roles of an instance, use the :ref:`roles <configuration_reference_roles>` option.
+
+    **Example: Built-in role**
+
+    The example below shows how to enable and configure the ``roles.crud-router`` role provided by the `CRUD <https://github.com/tarantool/crud>`__ module:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster_crud/config.yaml
+        :language: yaml
+        :start-at: roles.crud-router
+        :end-at: stats_quantile_max_age_time
+        :dedent:
+
+    You can find the full example here: `sharded_cluster_crud <https://github.com/tarantool/doc/tree/latest/doc/code_snippets/snippets/sharding/instances.enabled/sharded_cluster_crud>`_.
+
+    **Example: Custom role**
+
+    The example below shows how to enable the custom ``greeter`` role for ``instance001`` and specify the configuration for this role:
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application_role/config.yaml
+        :language: yaml
+        :start-at: instance001
+        :end-at: greeting
+        :dedent:
+
+    You can define the ``validate()``, ``apply()``, and ``stop()`` functions in the role's code as follows:
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application_role/greeter.lua
+        :language: lua
+        :dedent:
+
+    You can find the full example here: `application_role <https://github.com/tarantool/doc/tree/latest/doc/code_snippets/snippets/config/instances.enabled/application_role>`_.
```

Comment:
It seems, it duplicates the same examples from the `roles` description. The size of this block is close to a viewport height in my case, so I feels this duplication as undesirable.

I would highlight here my idea to move these examples to a 'how to write a role' guide and link it from here.